lib/explorer/series.ex

```diff
@@ -5651,6 +5651,32 @@ defmodule Explorer.Series do
   def split(%Series{dtype: dtype}, _by),
     do: dtype_error("split/2", dtype, [:string])
 
+  @doc """
+  Split a string Series into a struct of string `fields`.
+  
+  The length of the field names list determines how many times the
+  string will be split at most. If the string cannot be split into that
+  many separate strings, null values will be provided for the
+  remaining fields.
+
+  ## Examples
+
+      iex> s = Series.from_list(["Smith, John", "Jones, Jane"])
+      iex> Series.split_into(s, ", ", ["Last Name", "First Name"])
+      #Explorer.Series<
+        Polars[2]
+        struct[2] [%{"First Name" => "John", "Last Name" => "Smith"}, %{"First Name" => "Jane", "Last Name" => "Jones"}]
+      >
+
+  """
+  @doc type: :string_wise
+  @spec split_into(Series.t(), String.t(), list(String.t() | atom())) :: Series.t()
+  def split_into(%Series{dtype: :string} = series, by, fields) when is_binary(by),
+    do: apply_series(series, :split_into, [by, fields])
+
+  def split_into(%Series{dtype: dtype}, _by, _fields),
+    do: dtype_error("split_into/3", dtype, [:string])
```

Comment:
```suggestion
  def split_into(%Series{dtype: :string} = series, by, [_ | _] = fields) when is_binary(by),
    do: apply_series(series, :split_into, [by, Enum.map(fields, &to_string/1)])

  def split_into(%Series{dtype: dtype}, by, [_ | _] = _fields)  when is_binary(by),
    do: dtype_error("split_into/3", dtype, [:string])
```

Reply:
Let's normalize fields to string before and make sure it is a non-empty list.

Reply:
@josevalim - The underlying polars `splitn` function takes a list of arbitrary length.  Do you think it makes sense to limit the explorer API to only use a list of two fields?  Alternatively, I could put a guard to ensure that `fields` is a list so that users can split the column however many times necessary.